packages/wallet/contract/src/smartWallet.js

```diff
@@ -1,90 +1,273 @@
 // @ts-check
+import { AmountMath } from '@agoric/ertp';
 import {
   makePublishKit,
   makeStoredSubscriber,
   observeIteration,
 } from '@agoric/notifier';
+import { makeScalarWeakMapStore } from '@agoric/store';
+import { makeScalarBigMapStore } from '@agoric/vat-data';
 import { E, Far } from '@endo/far';
-import { makeWallet } from './support.js';
+import { makeInvitationsHelper } from './invitations.js';
+import { makeOffersFacet } from './offers.js';
+import { paymentsFromPurses, paymentsIntoPurses } from './utils.js';
 
-const { assign, entries, keys, fromEntries } = Object;
+/**
+ * @file Smart wallet module
+ *
+ * Design notes:
+ * Money comes in: 1) on depositFacet, 2) proceeds/payouts of an offer
+ * Money goes out: ONLY by making offers
+ *
+ * Requirements:
+ * State must not grow too large.
+ * If JS runtime dies no assets are lost.
+ *   ??? what happens if you withdraw $10 from your purse to a payment for your offer and JS dies before you take payouts?
+ */
+
+/** @typedef {Array<BrandDescriptor & {purseBalance: Amount}>} PursesRecord */
+
+/**
+ * Table of offer IDs to their statuses. This is not exhaustive of all offers because that would grow the state monotonically and eventually to excees.
+ *
+ * Each publication is a subset of the virtual table of all offers. Currently, each publication is for only one offer.
+ *
+ * @typedef {{ [offerId: number]: import('./offers.js').OfferStatus }} OffersRecord
+ */
 
 /**
+ * @typedef {{
+ * brand: Brand,
+ * issuer: ERef<Issuer>,
+ * petname: import('./types').Petname
+ * }} BrandDescriptor
+ */
+
+/**
+ *
  * @param {{
  * address: string,
  * bank: ERef<import('@agoric/vats/src/vat-bank').Bank>,
- * myAddressNameAdmin: ERef<MyAddressNameAdmin>,
  * }} unique
  * @param {{
- * agoricNames: ERef<NameHub>,
  * board: ERef<Board>,
- * namesByAddress: ERef<NameHub>,
+ * invitationIssuer: ERef<Issuer<'set'>>,
  * storageNode: ERef<StorageNode>,
  * zoe: ERef<ZoeService>,
  * }} shared
  */
 export const makeSmartWallet = async (
-  { address, bank, myAddressNameAdmin },
-  { agoricNames, board, namesByAddress, storageNode, zoe },
+  { address, bank },
+  { board, invitationIssuer, storageNode, zoe },
 ) => {
   assert.typeof(address, 'string', 'invalid address');
   assert(bank, 'missing bank');
-  assert(myAddressNameAdmin, 'missing myAddressNameAdmin');
+  assert(invitationIssuer, 'missing invitationIssuer');
   assert(storageNode, 'missing storageNode');
+  const invitationBrand = await E(invitationIssuer).getBrand();
+  const marshaller = await E(board).getReadonlyMarshaller();
 
-  const cacheStorageNode = E(storageNode).makeChildNode('cache');
+  // #region precious state
+  /**
+   * Private purses
+   *
+   * @type {MapStore<Brand, ERef<Purse<'set'>>>}
+   */
+  const brandPurses = makeScalarBigMapStore('brand purses', { durable: true });
```

Comment:
It still seems like we could reduce the scope of this by, for example, moving it along with `publishPurses` later in the file.

Reply:
Good idea. For auditability I tried to gather the state and for upgradability I tried to gather precious state. But it would be better to have a comment explicitly about state requirements and let them be defined in the most advantageous place for safety.